src/test/groovy/graphql/schema/validation/TypeAndFieldRuleTest.groovy

```diff
@@ -0,0 +1,123 @@
+package graphql.schema.validation
+
+import graphql.AssertException
+import graphql.TestUtil
+import graphql.schema.GraphQLNonNull
+import graphql.schema.GraphQLObjectType
+import spock.lang.Specification
+
+
+import static graphql.Scalars.GraphQLString
+import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition
+import static graphql.schema.GraphQLObjectType.newObject
+import static graphql.schema.GraphQLSchema.newSchema
+
+class TypeAndFieldRuleTest extends Specification {
+
+
+    def "type must define one or more fields."() {
```

Comment:
we should add tests for Interfaces here as well.

Reply:
done

Reply:
Thanks for your suggestion.